pyiron_gui/project/project_browser.py

```diff
@@ -30,38 +31,38 @@
 __date__ = "Feb 02, 2021"
 
 
-class PyironWrapper:
+@singledispatch
+def PyironWrapper(py_obj, project, rel_path=""):
+    return BaseWrapper(py_obj, project, rel_path=rel_path)
 
-    """Simple wrapper for pyiron objects which extends for basic pyiron functionality (list_nodes ...)"""
 
-    _with_self_representation = {
-        "structure": Atoms,
-        "murnaghan": Murnaghan
-    }
+@PyironWrapper.register
+def _(py_obj: Atoms, project, rel_path=""):
+    return AtomsWrapper(py_obj, project, rel_path=rel_path)
+
+
+@PyironWrapper.register
+def _(py_obj: Murnaghan, project, rel_path=""):
+    return MurnaghanWrapper(py_obj, project, rel_path=rel_path)
+
+
+class BaseWrapper:
+
+    """Simple wrapper for pyiron objects which extends for basic pyiron functionality (list_nodes ...)"""
 
     def __init__(self, pyi_obj, project, rel_path=""):
         self._wrapped_object = pyi_obj
         self._project = project
-        if not hasattr(pyi_obj, 'path'):
-            self._path = self._project.path
         self._rel_path = rel_path
-        # print("init:" + self.path)
-        self._type = None
-        for name, cls in self._with_self_representation.items():
-            if isinstance(pyi_obj, cls):
-                self._type = name
-
-    @property
-    def has_self_representation(self):
-        return self._type is not None
+        self._name = None
 
     @staticmethod
```

Comment:
Can be inlined, since it's only used once.

Reply:
Please teach me how 😅 
I use this `_empty_list` right now in 
```   
def __getattr__(self, item):
        if item in ['list_nodes', 'list_groups']:
            try:
                return getattr(self._wrapped_object, item)
            except AttributeError:
                return self._empty_list
        return getattr(self._wrapped_object, item)
```
where I want to return a callable with no arguments which returns an empty list. I thought about just returning `list`, however, if this is at some point called with some arguments, it will return something else than `[]`.

Reply:
A simple way would be 
```python
def __getattr__(self, item):
        if item in ['list_nodes', 'list_groups']:
            try:
                return getattr(self._wrapped_object, item)
            except AttributeError:
                return lambda: [] 
        return getattr(self._wrapped_object, item)
```
I had actually missed that `__getattr__` is dispatching to the wrapped `list_nodes()`/`list_groups()`.  In this case I have a slight preference for actually subclassing `BaseWrapper` off `HasGroups` and implementing it explicitly.